src/main/java/com/epam/reportportal/auth/endpoint/SamlConfigurationEndpoint.java

```diff
@@ -0,0 +1,183 @@
+import static com.epam.ta.reportportal.commons.Predicates.not;
+import static com.epam.ta.reportportal.commons.validation.BusinessRule.expect;
+import static java.util.Optional.ofNullable;
+
+/**
+ * This class is responsible for provisioning REST API for managing of SAML identity providers' settings
+ *
+ * @author Yevgeniy Svalukhin
+ */
+@Controller
+@RequestMapping("/settings/saml")
+public class SamlConfigurationEndpoint {
+
+	private final ApplicationEventPublisher eventPublisher;
+	private SamlProviderProvisioning<ServiceProviderService> serviceProviderProvisioning;
+	private SamlProviderDetailsRepository repository;
+
+	public SamlConfigurationEndpoint(ApplicationEventPublisher eventPublisher,
+			SamlProviderProvisioning<ServiceProviderService> serviceProviderProvisioning, SamlProviderDetailsRepository repository) {
+		this.eventPublisher = eventPublisher;
+		this.serviceProviderProvisioning = serviceProviderProvisioning;
+		this.repository = repository;
+	}
+
+	@GetMapping("/{providerName}")
+	@ResponseBody
+	@ResponseStatus(HttpStatus.OK)
+	@Transactional(readOnly = true)
+	public SamlDetailsResource getSamlSettingsByName(@PathVariable String providerName) {
+		SamlProviderDetails samlProviderDetails = repository.findByIdpName(providerName)
+				.orElseThrow(() -> new ReportPortalException(ErrorType.OAUTH_INTEGRATION_NOT_FOUND, providerName));
+		return TO_RESOURCE.apply(samlProviderDetails);
+	}
+
+	@GetMapping
+	@ResponseBody
+	@ResponseStatus(HttpStatus.OK)
+	@Transactional(readOnly = true)
+	public List<SamlDetailsResource> getAllSamlSettings() {
+		return repository.findAll().stream().map(TO_RESOURCE).collect(Collectors.toList());
+	}
+
+	@DeleteMapping("/{providerId}")
+	@ResponseBody
+	@ResponseStatus(HttpStatus.OK)
+	@Transactional
+	public OperationCompletionRS deleteSamlSetting(@PathVariable Long providerId) {
+		SamlProviderDetails samlProviderDetails = repository.findById(providerId)
+				.orElseThrow(() -> new ReportPortalException(ErrorType.OAUTH_INTEGRATION_NOT_FOUND, providerId));
+		repository.delete(samlProviderDetails);
+		eventPublisher.publishEvent(new SamlProvidersReloadEvent(repository.findAll()));
+		return new OperationCompletionRS("Auth settings '" + samlProviderDetails.getIdpName() + "' were successfully removed");
+	}
+
+	@PostMapping
+	@ResponseBody
+	@ResponseStatus
+	@Transactional
+	public SamlDetailsResource createSamlSettings(@RequestBody @Validated SamlDetailsResource samlDetailsResource) {
+		expect(repository.findByIdpName(samlDetailsResource.getIdentityProviderName()),
+				not(isPresent())
+		).verify(ErrorType.INTEGRATION_ALREADY_EXISTS, samlDetailsResource.getIdentityProviderName());
+		validate(samlDetailsResource);
+		SamlProviderDetails samlProviderDetails = populateProviderDetails(samlDetailsResource);
+		repository.save(samlProviderDetails);
+		eventPublisher.publishEvent(new SamlProvidersReloadEvent(repository.findAll()));
```

Comment:
why do we need findAll here?

Reply:
We have one handler for create, update and delete actions. It updates whole list of providers